app/static/js/index.js

```diff
@@ -0,0 +1,58 @@
+// Function to check if an element is in the viewport
+let inViewId = "";
+
+const inView = (element) => {
```

Comment:
Here you could probably also use the more modern [IntersectionObserver API](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/observe)?